src/plugins/controls/public/options_list/types.ts

```diff
@@ -21,7 +22,8 @@ interface SearchString {
 
 // Component state is only used by public components.
 export interface OptionsListComponentState {
-  field?: OptionsListField;
+  popoverOpen: boolean;
```

Comment:
I've made the status of the popover part of the component state so that I can ensure that it is closed both (a) when the component is unmounted and (b) when there is a fatal error.